### PR TITLE
API Sorts and Filters

### LINKED ISSUE #10: API endpoints should always be ordered and filtered correctly
All `index` API endpoints should order the results as expected (by default). This also needs to account for filtering for example, hidden component groups.

### PR DESCRIPTION
**Pull request by joelbutcher**
This PR updates `MetricController::index()` and `IncidentController::index()` to default the ordering of results by the `created_at` date in descending order.

Tests have been included to check this is the case, and to check the default sorting, the ability to sort and the ability to filter various resources.

Resolves #10

### CODE REVIEW

File: tests/TestCase.php [hunk 1]
```diff
@@ -18,6 +19,8 @@ protected function setUp(): void
         Factory::guessFactoryNamesUsing(
             fn (string $modelName) => 'Cachet\\Database\\Factories\\'.class_basename($modelName).'Factory'
         );
+
+        $this->withoutApiRateLimiting();
```
Comment (joelbutcher) [inline tests/TestCase.php:23]: This has been added because running `composer test` ends up hitting the rate limit for the `api` guard set by workbench 😢

Comment (jbrooksuk) [reply tests/TestCase.php:23]: @joelbutcher I ran into this myself and I wonder whether we should create a custom Rate Limit for Cachet?

Comment (joelbutcher) [reply tests/TestCase.php:23]: Can do? Could have it configurable to developers via the cachet config file, too

Comment (jbrooksuk) [reply tests/TestCase.php:23]: We could set a custom middleware in the `api_middleware` config.

**Review by jbrooksuk — CHANGES_REQUESTED** [review]
Can we bring back some of the tests that were removed?

File: tests/TestCase.php (hunk 1 above)
Comment (joelbutcher) [reply tests/TestCase.php:23]: @jbrooksuk Would be better to separate that from this PR as it's not a blocker for what we're adding here, but would be good to include regardless

File: tests/Feature/Api/ComponentGroupTest.php [hunk 2]
```diff
@@ -26,13 +26,14 @@
     $response->assertJsonCount(15, 'data');
 });
 
-it('can list more than 15 components', function () {
```
[the remark this answers is not in the document]
Comment (joelbutcher) [reply tests/Feature/Api/ComponentGroupTest.php:29]: Yup, my bad 😅

**Pull request merged**